src/modules/facility-loan-transaction/dto/get-loan-transaction-response.dto.ts

```diff
@@ -0,0 +1,158 @@
+    example: EXAMPLES.PRODUCT_TYPE_GROUP,
+  })
+  readonly productTypeGroup: string;
+
+  @ApiProperty({ description: 'The currency code for the loan.', minLength: 3, maxLength: 3, example: EXAMPLES.CURRENCY })
+  readonly currency: string;
+
+  @ApiProperty({
+    description:
+      'The exchange rate between the loan currency and the deal currency. It is only applicable when the loan currency differs from the facility currency.',
+    example: EXAMPLES.DEAL_CUSTOMER_USAGE_RATE,
+  })
+  readonly dealCustomerUsageRate: number | null;
+
+  @ApiProperty({
+    description:
+      'This represents the operand (M for multiply or D for divide) used to convert the amounts. It is only applicable when the loan currency differs from the facility currency.',
+    minLength: 0,
+    maxLength: 1,
+    example: EXAMPLES.DEAL_CUSTOMER_USAGE_RATE,
+  })
+  readonly dealCustomerUsageOperationType: string | null;
+
+  @ApiProperty({
+    description: 'The amount of the loan. It is called loan amount in ACBS.',
+    example: EXAMPLES.LOAN_AMOUNT,
+  })
+  readonly amount: number;
+
+  @ApiProperty({
+    description: 'The issue date for the loan. It is called the effective date in ACBS.',
+    example: EXAMPLES.DATE_ONLY_STRING,
+    type: Date,
+    format: 'date',
+  })
+  readonly issueDate: DateOnlyString;
+
+  @ApiProperty({
+    description: 'The expiry date for the loan. It is called the maturity date in ACBS.',
+    example: EXAMPLES.DATE_ONLY_STRING,
+    type: Date,
+    format: 'date',
+  })
+  readonly expiryDate: DateOnlyString;
+
+  @ApiProperty({
+    description:
+      'The spread rate of the PAC accrual schedule to factor into the all-in rate if a rate calculation method is selected that includes the spread rate in the calculation.',
+    example: EXAMPLES.SPREAD_RATE,
+  })
+  readonly spreadRate: number;
+
+  @ApiProperty({
+    description:
+      'The spread rate of the CTL accrual schedule to factor into the all-in rate if a rate calculation method is selected that includes the spread rate in the calculation.',
+    example: EXAMPLES.SPREAD_RATE,
+  })
+  readonly spreadRateCTL: number;
+
+  @ApiProperty({
+    description: 'A code denoting the year basis for the accrual schedule.',
+    minLength: 0,
+    maxLength: 1,
+    example: EXAMPLES.YEAR_BASIS,
+  })
+  readonly yearBasis: string;
+
+  @ApiProperty({
+    description: 'The date the next payment will be due for the loan repayment schedule.',
+    example: EXAMPLES.DATE_ONLY_STRING,
+    type: Date,
+    format: 'date',
+  })
+  nextDueDate: DateOnlyString;
+
+  @ApiProperty({
+    description: `A code denoting the index rate change frequency, which is used by ACBS to determine the frequency at which the rate should change when the change timing is set to 'On Anniversary'.`,
+    minLength: 0,
+    maxLength: 1,
+  })
+  indexRateChangeFrequency: string;
```

Comment:
do we need these properties (including the one below) to be readOnly?

Reply:
Changed.